engine/src/org/pentaho/di/trans/steps/tableoutput/TableOutput.java

```diff
@@ -303,7 +303,7 @@ public boolean processRow( StepMetaInterface smi, StepDataInterface sdi ) throws
       }
 
       // See if we need to get back the keys as well...
-      if ( meta.isReturningGeneratedKeys() ) {
+      if ( meta.isReturningGeneratedKeys() && meta.getDatabaseMeta().supportsAutoGeneratedKeys() ) {
```

Comment:
For table output here, is it possible to return generated keys without supporting autogenerated keys?  Could we maybe put the check for whether the dialect supports it into the meta and incorporate that into the isReturningGeneratedKeys() call?